server/src/main/java/org/elasticsearch/cluster/coordination/PeerFinder.java

```diff
@@ -102,10 +102,8 @@ public void activate(final DiscoveryNodes lastAcceptedNodes) {
     public void deactivate(DiscoveryNode leader) {
         synchronized (mutex) {
             logger.trace("deactivating PeerFinder and setting leader to {}", leader);
-            if (active) {
-                active = false;
-                handleWakeUp();
-            }
+            active = false;
```

Comment:
The tests deactivate the `PeerFinder` at the end, regardless of whether it's active or not, and there's no particular reason to avoid double-deactivation or to make sure that each test ends with an active `PeerFinder`.

Also although I am pretty sure that we can't change the leader without being active, this is not something we guarantee, nor do I think we need to.